dashboard/dashboard.py

```diff
@@ -223,18 +229,29 @@ async def run(self):
         loop = asyncio.get_event_loop()
         loop.run_until_complete(dashboard.run())
     except Exception as e:
-        # Something went wrong, so push an error to all drivers.
-        redis_client = ray._private.services.create_redis_client(
-            args.redis_address, password=args.redis_password)
         traceback_str = ray._private.utils.format_error_message(
             traceback.format_exc())
         message = f"The dashboard on node {platform.uname()[1]} " \
                   f"failed with the following " \
                   f"error:\n{traceback_str}"
-        ray._private.utils.push_error_to_driver_through_redis(
-            redis_client, ray_constants.DASHBOARD_DIED_ERROR, message)
         if isinstance(e, FrontendNotFoundError):
             logger.warning(message)
         else:
             logger.error(message)
             raise e
+
+        # Something went wrong, so push an error to all drivers.
+        redis_client = ray._private.services.create_redis_client(
+            args.redis_address, password=args.redis_password)
+        gcs_publisher = None
+        if args.gcs_address:
+            gcs_publisher = gcs_utils.GcsPublisher(address=args.gcs_address)
+        elif os.environ.get("RAY_gcs_grpc_based_pubsub") == "true":
```

Comment:
```suggestion
        elif os.environ.get("RAY_gcs_grpc_based_pubsub") not in ["0", "false"]:
```

Slight preference to bias to True here.

Reply:
Moved to a helper method with similar refactoring.